### ISSUE TITLE
Black Screen after an In-Game Trade in Pokemon White Version 2

### ISSUE THREAD

**Issue by Priestyle** [melonDS-emu/melonDS#1719]
In the game Pokemon White Version 2, I get a black screen after an in-game trade with the NPC Diana in Accumula Town. So I tried to do the same story event on my game with TWiLight Menu++ and nds-bootstrap but I didn't get a black screen. I initially thought this was a problem with my save file but it turned out not to be. I used MelonDS PoroCYon synchronize #1583 PoroCYon:gdbstub-standalone Windows Build. I just want to add at the end that the game treats this in-game NPC trade as an IR trade.

**Comment by RSDuck** [comment]
can you give us a savefile for reproduction?

**Comment by KostaSaizo7** [comment]
Maybe this [Anti-Piracy](https://tcrf.net/Pok%C3%A9mon_Black_and_White#Anti-Piracy) thing is related?

**Comment by Blazekickn** [comment]
Saved in front of the Mantine for Tangrowth trade with a Mantine in the party: [White 2 Save File](https://cdn.discordapp.com/attachments/443617032748007451/1178870057871343756/Subset_Pokemon_White_Version_2_Professor_Oak_Challenge.srm?ex=6577b74c&is=6565424c&hm=6748c9ab9652ce34118623ebd2ac3d447ce783f93fc96dad9cfec9d2dae97684&)

I originally had this issue with the melonds-ds fork but I guess it's in the standalone as well, so hopefully this helps.

**Comment by KostaSaizo7** [comment]
I tried with 1.0 RC and the issue is (somehow?) fixed

**Comment by LudixKuroken** [comment]
Hello. Happened the same to me with the in-game trade in Route 4.
I do the trade and when it finishes the screen goes black.

To give some more background, I'm using MelonDS through the steam Reatroarch launcher, with the Pokémon White 2 (USA, Europe) version of the game, although it is patched to unlock the unova link and the keys.

Tried recreating it with the latest version of MelonDs without Retroarch and everything worked, but felt like I needed to share.

**Comment by naqi-kun** [comment]
I used the MelonDS emulator on android. What worked for me was just using the rewind function.

**Comment by SNameless1** [comment]
> Exact same thing happened to me today when i tried to trade on Route 4. I tried to switch on and off different settings, and somehow fixed it by turning off the JIT recompiler.

If anyone is still wondering how can fix that issue, this worked for me too.

**Comment by OctopusButtons** [comment]
Yes, I'm using latest melon**DS DS** core in RetroArch 1.21.0, and the Route 4 NPC trade (house on right/east side) crashes to black screen while music is still playing but no sound effects or apparent response to input.  **Turning Off JIT** in core options, then quitting game and restarting, made the trade work fine.

**Comment by V-Karch** [comment]
Can confirm I'm having the same issue just using melonds android from the google play store; solved by disabling JIT temporarily and then enabling it after the trade
